Comments on master branch.

MarkHaroldson opened the pull request:
I added some comments to the code.

Generally everything looks good, but the real work happens when you start to incorporate these functions into a feature rich system,

MarkHaroldson committed:
Comments on master branch.

MarkHaroldson committed:
Added a WPF UX on to original project + refactored KeyManager.

MarkHaroldson commented:
I added a quick WPF application to the the original project.

Also, I noticed a crash when encrypting a large (1k+) text using RSA PKCS. @Theodor488  Can you investigate what is going on.

Theodor488 committed:
Resolving "Why use "out" variables rather than create an simple class to contain public/private key?"

Theodor488 committed:
merge

Theodor488 committed:
merge conflict

MarkHaroldson commented on SecureMessageTransmitter/RsaKeyPair.cs:
```diff
@@ -0,0 +1,20 @@
+
+namespace SecureMessageTransmitter
+{
+    public class RsaKeyPair
```
This change breaks SMT-UX app.

Theodor488 replied:
Latest commit fixes the UX app

Theodor488 committed:
Fixed change that breaks SMT-UX app. There was a conflict between KeyPair.cs and Models.keyPair.cs.

Theodor488 committed:
- Added Aes Encryption/Decryption.
- Added logic to encrypt AES key with RSA and then use the AES key for longer message encryption/decryption

Theodor488 committed:
- Fixed ASn.1 inconsistent tag caused by incorrect AES encryption. Originally in ASES encryption I was using StreamWriter which is not correct for byte array data because StreamWriter is used for writingchars and caused encoding issues. Taking byte array directly from CryptoStream is correct.
- Added logging for each step
